test/NuGet.Core.Tests/NuGet.Protocol.Tests/Resources/VulnerabilityInfoResourceV3Tests.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Model;
+using NuGet.Protocol.Resources;
+using NuGet.Protocol.Tests.Providers;
+using NuGet.Test.Utility;
+using Test.Utility;
+using Xunit;
+
+namespace NuGet.Protocol.Tests.Resources
+{
+    public class VulnerabilityInfoResourceV3Tests
```

Comment:
We could use a few more tests (can be in a future PR). 

- GetVulnerabilityDataAsync 
  - Fetches data from all index entries (testing more than 1).
  - Multiple calls use http cache.
  - Multiple calls when 1 file gets updated only fetches the updated file.